### PR TITLE
searching stops

### PR DESCRIPTION
**Pull request by petermarathas**
Added the option to search on the Stops page, and the Search page now does not display items until a search query is typed. Also added a nice centered message for all three pages when results are not found.

:bowtie:

### CODE REVIEW

**Comment by akaplo** [comment]
This looks good! I pulled it and played around with it and it works as it should.  

Once you resolve the merge conflicts, it should be good to go.  

There's also a weird thing where Search is now dropping a couple of routes from the list when calling prepareRoutes.  I don't think that it was introduced by your code, but I noticed it while testing it so I thought I'd bring it up.

**Comment by petermarathas** [comment]
We should make sure that the routes thing isn't just a problem with this branch before we merge. 

Also, I realized that there was a problem with my code - the stops controller worked when the stops weren't loaded yet, but what I had written before didn't work when the stops were loaded somewhere else (e.g. the search page). Took me some time but now I see: StopList stores the stops as a hash where the StopId is the key, but the stops controller and search controller need to convert this loaded hash into an array in order to filter. 

Long story short searching for stops works for real now.
